backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -92,6 +94,31 @@
                 .onItem().call(this::loadProperties);
     }
 
+    public Uni<Endpoint> getOrCreateEmailSubscriptionEndpoint(String accountId, EmailSubscriptionProperties properties) {
+        return getEndpointsPerType(accountId, EndpointType.EMAIL_SUBSCRIPTION, true, null)
+                .onItem().call(this::loadProperties)
+                .onItem().transformToUni(emailEndpoints -> {
+                    Optional<Endpoint> endpointOptional = emailEndpoints
+                            .stream()
+                            // Todo: This should be changed once we store the properties - (properties = null right now)
+                            // .filter(endpoint -> endpoint.getProperties().equals(properties))
```

Comment:
Related with my comment on `EmailSubscriptionProperties`. Will that work knowing that properties equality is currently based on the DB primary key?

Reply:
Currently we never load/save `EmailSubscriptionProperties`, it's always null.

Reply:
but when we load/save them, as the received `properties` is not in the database, it doesn't have any `id` and we need to check on the contents.

Reply:
Can we simply use a new method for the comparison and keep the existing `equals`/`hashCode` methods unchanged? It could be something like `hasSameProperties` or `equalsProperties` that would only be used for that use case.

I am really concerned about overriding `equals`/`hashCode` for that purpose. It could have bad consequence in the near future if we put properties in a `Set` or even with Hibernate. I know the non-reactive version of Hibernate requires proper `equals`/`hashCode` implementations in some cases (that we are not using in notifications) but I am really unsure about Hibernate Reactive.

Reply:
Sure, I can create a method to compare `EmailSubscriptionProperties` and leave the equals / hasCode alone

Reply:
Updated